pennylane/circuit_graph.py

```diff
@@ -12,305 +12,278 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-CircuitGraph
-============
+Circuit graph
+=============
 
 **Module name:** :mod:`pennylane.circuit_graph`
 
 .. currentmodule:: pennylane.circuit_graph
 
 This module contains the CircuitGraph class which is used to generate a DAG (directed acyclic graph)
-based on operation and observable queues.
+representation of a quantum circuit.
 
-.. raw:: html
 
-    <h3>Summary</h3>
+Classes
+-------
 
 .. autosummary::
-    CircuitGraph
+   CircuitGraph
+   Layer
 
-.. raw:: html
 
-    <h3>Code details</h3>
+CircuitGraph methods
+--------------------
+
+.. currentmodule:: pennylane.circuit_graph.CircuitGraph
+
+.. autosummary::
+   operations
+   observables
+   graph
+   wire_indices
+   ancestors
+   descendants
+   ancestors_in_order
+   descendants_in_order
+   _in_topological_order
+   layers
+   iterate_layers
+   update_node
+
+.. currentmodule:: pennylane.circuit_graph
+
+Code details
+~~~~~~~~~~~~
 """
 from collections import namedtuple
 import networkx as nx
 
 
-Command = namedtuple("Command", ["name", "op", "return_type", "idx"])
-Layer = namedtuple("Layer", ["op_idx", "param_idx"])
+Layer = namedtuple("Layer", ["ops", "param_inds"])
+"""TODO Parametrized layer of the circuit."""
 
 
 class CircuitGraph:
-    """Represents a queue of operations and observables
-    as a directed acyclic graph.
+    """Represents a quantum circuit as a directed acyclic graph.
 
     Args:
-        queue (list[Operation]): the quantum operations to apply
-        observables (list[Observable]): the quantum observables to measure
-        parameters (dict[int->list[(int, int)]]): Specifies the free parameters
-            of the quantum circuit. The dictionary key is the parameter index.
-            The first element of the value tuple is the operation index,
+        ops (Iterable[Operation]): quantum operations constituting the circuit, in temporal order
+        variable_deps (dict[int, list[ParDep]]): Free parameters of the quantum circuit.
+            The dictionary key is the parameter index.
+            The first element of the ParDep namedtuple is the operation index,
```

Comment:
Does linking to `ParDep` like so work?
```suggestion
            The first element of the :class:`~.ParDep` namedtuple is the operation index,
```

Reply:
Yes, but the ParDep in the type field already is a link.